apps/model_eval_viz/app.py

```diff
@@ -159,9 +157,38 @@ def display_states(state_list:list=state_selector.value)->gv.Polygons:
     
 # create a DynamicMap to allow Panel to link state_selector with a Geoviews(Holoviews under the hood) object
 # replaces @pn.depends
-displayed_states = hv.DynamicMap(pn.bind(display_states, state_list=state_selector))
+displayed_states = pn.rx(display_states)(state_selector)
 
-def display_points(state_list:list=state_selector.value)->gv.Points:
+
+
+def enter_event(event)->None:
+    '''
+    Event handler function for the 'Enter' button widget.
+
+    Checks if the value entered in the 'streamgage_input' widget contains any non-numeric or non-comma characters. If so, clears the input value of 'streamgage_input',
+    clears the value of 'entered_points' widget and sets the placeholder text for 'streamgage_input'. If the input value is valid, updates the value of 'entered_points'
+    widget to match the value of 'streamgage_input'.
+    
+    Parameters:
+    event : The input in the textbox for the data set 
+
+    Returns: None
+    '''
+    sg_value = streamgage_input.value 
+    if re.search('[^0-9, ]',sg_value):
+        streamgage_input.value = '' 
+        entered_points.value = ''
+        streamgage_input.placeholder ="Expected Format:[id_no], [id_no2]..."
+    else:
+        entered_points.value = sg_value
+enter_id = pn.panel(pn.widgets.Button(name='Enter', button_type='primary'))
+
+enter_id.on_click(enter_event)
+
+
+
+
+def display_points(state_list:list=state_selector.value,ids:str=entered_points.value, data_set:str=subset_selector.value)->gv.Points:
```

Comment:
I beleive you can delete all of the default values here as these will be passed by the pn.rx() function. 

You can then update data_set = data_set, as that will be passed by pn.rx().